src/filter/macros.rs

```diff
@@ -0,0 +1,139 @@
+#[macro_export]
+// macro to initialize vectors with a given capacity to reduce copy / paste
+macro_rules! store_vec {
+    // $name: the name of the variable
+    // $type: Type of data stored in vector (Mat5 / Vec5)
+    // $capacity: How much space to allocate
+    ($capacity:expr ; $($name:ident : $type:ty),+ ) => {
+        $(
+            let mut $name : Vec<$type> = Vec::with_capacity($capacity);
+        )+
+        // let mut $name: Vec<$type> = Vec::with_capacity($capacity);
+    };
+}
+
+
+#[macro_export]
+/// Push a value into an iterator. Saves repeating vec.push(asdad)
+/// Easier to read this way.
+macro_rules! push {
+    ($($item:expr => $location:ident),*) => {
+        $(
+            $location.push($item);
+        )*
+    };
+    (remove: $index:expr; $($pop_vec:ident => $dest_vec:ident),+) =>{
+        $(
+            let removed_val = $pop_vec.remove($index);
+            push!{removed_val => $dest_vec};
+        )+
+    };
+    (remove: $index:expr;$($pop_vec:ident),+ ) => {
+        $($pop_vec.remove($index);)+
+    };
+}
+
+#[macro_export]
+/// Get a reference to an item in a vector using .get_unchecked(_)
+macro_rules! get_unchecked {
+    ($index:expr; $($vector:ident => $destination:ident),+) => {
+        $(
+            let $destination = 
+                unsafe {
+                    $vector.get($index).expect("get_unchecked!{} fetched something out of bounds")
+                }
+        )+
+    };
+}
+
+#[macro_export]
+/// Fetch the lengths of all iterators passed in. Used for debug
+macro_rules! length {
+    ($($var:ident),+) => {
+         $(
+             let coll: Vec<_> = $var.collect();
+             dbg!{coll.len()};
+         )+
+    };
+}
+
+
+#[macro_export]
+/// Used for getting the next value of the iterator. Additionally, this macro allows
+/// us to move the ownership of the data to "stagger" it. This does the following:
```

Comment:
I _think_ I mean iterate pairwise. The idea is that once we move to the next sensor the `$current` variable would become the `$previous`, the `$next` became `$current`, and so on. 
This macro was used heavily when originally working with iterators since there was a lot of repetitive code in the smoothing phase. I left it in since i expect to be working with iterators in the future. 
The replacement to this (which gives more control) is `get_unchecked!`, which _should_ have the same performance as the older iterator code i was working with.